pkg/scheduler/framework/plugins/noderesources/fit.go

```diff
@@ -44,11 +45,46 @@ const (
 	preFilterStateKey = "PreFilter" + FitName
 )
 
+// nodeResourceStrategyTypeMap maps strategy to scorer implementation
+var nodeResourceStrategyTypeMap = map[config.ScoringStrategyType]scorer{
+	config.LeastAllocated: func(args *config.NodeResourcesFitArgs) *resourceAllocationScorer {
+		resToWeightMap := resourcesToWeightMap(args.ScoringStrategy.Resources)
+		return &resourceAllocationScorer{
+			Name:                LeastAllocatedName,
+			scorer:              leastResourceScorer(resToWeightMap),
+			resourceToWeightMap: resToWeightMap,
+		}
+	},
+	config.MostAllocated: func(args *config.NodeResourcesFitArgs) *resourceAllocationScorer {
+		resToWeightMap := resourcesToWeightMap(args.ScoringStrategy.Resources)
+		return &resourceAllocationScorer{
+			Name:                MostAllocatedName,
+			scorer:              mostResourceScorer(resToWeightMap),
+			resourceToWeightMap: resToWeightMap,
+		}
+	},
+	config.RequestedToCapacityRatio: func(args *config.NodeResourcesFitArgs) *resourceAllocationScorer {
+		resToWeightMap := resourcesToWeightMap(args.ScoringStrategy.Resources)
+		return &resourceAllocationScorer{
+			Name:                RequestedToCapacityRatioName,
+			scorer:              requestedToCapacityRatioScorer(resToWeightMap, args.ScoringStrategy.RequestedToCapacityRatio.Shape),
+			resourceToWeightMap: resToWeightMap,
+		}
+	},
+}
```

Comment:
Alternatively, you can define a map which map from `ScoringStrategyType` to `scorer` directly, and initialize the `resourceAllocationScorer` at the end, this will help to reduce lot of duplicated code here.

Reply:
I don't see too much duplication, it is simple instantiation, I think this is fine as is.